src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,133 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INCREMENT = 2;
+    private int size;
+    private int threshold;
+    private int capacity;
+    private Node<K, V>[] table;
 
     @Override
     public void put(K key, V value) {
-
+        if (size == 0) {
+            initialization();
+        }
+        if (size == threshold) {
+            resize();
+        }
+        putValue(key, value);
     }
 
     @Override
     public V getValue(K key) {
+        if (table == null) {
+            return null;
+        }
+        int position = getIndex(key);
+        Node<K, V> tempNode = table[position];
+        int i = 0;
+        do {
+            if ((key != null
+                    && hash(tempNode.key) == hash(key)
+                    && key.equals(tempNode.key))
+                    || (key == null && tempNode.key == null)) {
+                return tempNode.value;
+            }
+            if (tempNode.next != null) {
+                tempNode = tempNode.next;
+            } else {
+                i = 1;
+            }
+        } while (i == 0);
```

Comment:
I really dont like this construction. Lets use smth like "while(tempNode != null)"